### PR TITLE
Improved performance of the delete confirmation dialog

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Fixes an issue for which deleting a single folder takes a long time to show the confirmation dialog

**Details of Changes**
Add details of changes here.
- When deleting a folder, only test the first file inside the folder to check if files can be sent to the recycle bin
- Small refactor of "Clipboard" FTP method

To test: try request deletion of the "Program Files" folder. Confirmation dialog should appear much quicker.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs [hunk 1]
```diff
@@ -821,6 +814,28 @@ await Win32API.SendMessageAsync(connection, new ValueSet()
...
+            }
+            using var shf = new ShellFolder(shi);
+            if (shf.FirstOrDefault(x => !x.IsFolder || x.Attributes.HasFlag(ShellItemAttribute.Stream)) is ShellItem item)
+            {
+                return item;
+            }
+            foreach (var shsfi in shf.Where(x => x.IsFolder && !x.Attributes.HasFlag(ShellItemAttribute.Stream)))
+            {
+                using var shsf = new ShellFolder(shsfi);
+                if (GetFirstFile(shsf) is ShellItem item2)
```
Comment (d2dyno1) [inline src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:831]: Looks cool! Would it be possible to do this using a loop instead of recursion?

**Review by d2dyno1 — APPROVED** [review]

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs (hunk 1 above)
Comment (gave92) [reply src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:831]: Probably, I understand that there's a programming theorem that says "every recursion be converted into iteration" :) In this case I find recursion more intuitive, but can be changed if you think it's necessary.

Comment (d2dyno1) [reply src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:831]: Ah, can be left as is :)

**Pull request merged**